src/components/button/code/templates/buttonTemplate.jsx

```diff
@@ -2,14 +2,15 @@ var React = require('react');
 var classNames = require('classnames');
 var getComponentClasses = require('../../../../utils/getComponentClasses');
 
-module.exports = function(props) {
+module.exports = function(context) {
```

Comment:
:+1: Agree with the alternative approach, will be much nicer to work with the same scope in the templates.